onnxruntime/core/graph/graph_proto_serializer.cc

```diff
@@ -43,12 +43,29 @@
   }
 
   if (include_initializer) {
+    std::unordered_set<std::string> current_scope_initializer_set;
+
     auto& initializers = graph_view.GetAllInitializedTensors();
     for (auto& it : initializers) {
       auto* p_initializer = graph_proto.add_initializer();
       *p_initializer = *(it.second);
+      current_scope_initializer_set.insert(it.first);
+    }
+
+    // handle outer scope value which is a constant initializer
+    for (auto& node_idx : graph_view.GetNodesInTopologicalOrder()) {
+      const auto& node = graph_view.GetNode(node_idx);
+      for (const auto& input : node->InputDefs()) {
+        if (current_scope_initializer_set.find(input->Name()) != current_scope_initializer_set.end()) {
+          continue;
+        }
+        if (graph_view.IsConstantInitializer(input->Name(), true)) {
+          auto* p_initializer = graph_proto.add_initializer();
+          *p_initializer = *(graph_view.GetConstantInitializer(input->Name(), true));
+        }
```

Comment:
What happens if the initializer is used by multiple nodes? Does it create duplicates?

Reply:
yes, need to check this.

Reply:
+1

Reply:
Already added the check.